### PR TITLE
Reduce App Router bundle size

### PR DESCRIPTION
**Pull request by blakewilson**
## Tasks

- [x] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.
- [x] If a code change, I have written testing instructions that the whole team & outside contributors can understand.
- [x] I have written and included a comprehensive [changeset](https://github.com/wpengine/faustjs/blob/canary/DEVELOPMENT.md#deployment) to properly document the changes I've made.

## Description

This PR reduces the bundle size of the experimental app router package by using the proper `ApolloClient` and `InMemoryCache` vs the [guide's ](https://www.apollographql.com/blog/announcement/frontend/using-apollo-client-with-next-js-13-releasing-an-official-library-to-support-the-app-router/#server-components) `NextSSRApolloClient`/`NextSSRInMemoryCache` utils. Reference [this issue](https://github.com/apollographql/apollo-client-nextjs/issues/95) for more context.

This PR also removes support for the CJS build as it is no longer needed with the ESM support in Next.js. 

## Testing

From the monorepo root:

1. `npm install`
2. `npm run build` to build the packages
3. Build the example project `npm run build -w examples/next/app-router`
4. Notice the bundle stats are 150kb+
5. Check out the `app-router-fix-bundle` branch
6. `npm install`
7. `npm run build`
8. Build the example project again `npm run build -w examples/next/app-router`
9. Notice the bundle stats are much more reasonable (~80kb)

## Screenshots

Before:
Screenshot 2023-09-18 at 8 26 31 AM

After:
Screenshot 2023-09-18 at 8 25 18 AM

### CODE REVIEW

File: packages/experimental-app-router/package.json [hunk 1]
```diff
@@ -2,25 +2,24 @@
   "name": "@faustwp/experimental-app-router",
   "version": "0.0.4",
   "description": "Experimental: A Faust package to support Next.js' App Router",
-  "main": "dist/cjs/index.js",
-  "module": "dist/mjs/index.js",
-  "types": "dist/cjs/index.d.ts",
+  "exports": {
+    ".": "./dist/index.js",
+    "./package.json": "./package.json"
```
Comment (blakewilson) [inline packages/experimental-app-router/package.json:7]: Package exports will come in handy once we start shipping `rsc` specific functionality vs `client` specific functions.

File: packages/experimental-app-router/src/client.ts [hunk 2]
```diff
@@ -65,8 +65,8 @@ async function createFaustApolloClient(authenticated = false) {
    * we may set config differently than how we currently do it.
    */
 
-  return new NextSSRApolloClient({
-    cache: new NextSSRInMemoryCache(inMemoryCacheObject),
+  return new ApolloClient({
+    cache: new InMemoryCache(inMemoryCacheObject),
```
Comment (blakewilson) [inline packages/experimental-app-router/src/client.ts:69]: This was what was blowing up the bundle size:

https://github.com/apollographql/apollo-client-nextjs/issues/95

**Review by theodesp — APPROVED** [review]
LGTM!

**Comment by blakewilson** [comment]
Looks like there is some issues with ESM and jest/tests. I'll get that worked out.

**Comment by blakewilson** [comment]
Getting Jest to play nice with just ESM is proving to be a pain. I'm going to take out the ESM related changes for now and we can revisit as needed. The spirit of this work is still present with using the different Apollo utils.

**Review by theodesp — APPROVED** [review]

**Pull request merged**